### PR TITLE
Add Filters Contingency lists

### PR DESCRIPTION
**Pull request by ChamseddineBhd**
Signed-off-by: chamseddine <chamseddine.benhamed@ensi-uma.tn>

### CODE REVIEW

File: src/main/java/org/gridsuite/actions/server/dto/FilterContingencyList.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    @ApiModelProperty("equipmentName")
+    private String equipmentName;
+
+    @ApiModelProperty("equipmentType")
+    private List<String> equipmentType;
+
+    @ApiModelProperty("nominalVoltage")
+    private String nominalVoltage;
+
+    @ApiModelProperty("nominalVoltageOperator")
```
Comment (geofjamg) [inline src/main/java/org/gridsuite/actions/server/dto/FilterContingencyList.java:41]: What is the purpose of this filter? What is the difference with nominalVoltage?

**Review by geofjamg — CHANGES_REQUESTED** [review]
The purpose of equipmentId and equipmentName is a regex filtering not strict filtering. What could be the purpose of a filter that just get one equipement? It does not make any sens, for that case we will develop another list type with explicit equipment configuration.

File: src/main/java/org/gridsuite/actions/server/dto/FilterContingencyListAttributes.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+/**
+ * @author Chamseddine benhamed <chamseddine.benhamed at rte-france.com>
+ */
+
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+public class FilterContingencyListAttributes {
```
Comment (geofjamg) [inline src/main/java/org/gridsuite/actions/server/dto/FilterContingencyListAttributes.java:24]: What is the difference with FilterContingencyList? Why 2 different classes?

Comment (ChamseddineBhd) [reply src/main/java/org/gridsuite/actions/server/dto/FilterContingencyListAttributes.java:22]: FilterContingencyListAttributes  is used as Body model for  the /filters-contingency-lists/{name}  PUT endpoint

File: src/main/java/org/gridsuite/actions/server/ContingencyListService.java [hunk 3]
```diff
@@ -36,69 +42,242 @@
...
+            case SHUNT_COMPENSATOR:
+                contingencies = getSCContingencyList(network, filtersContingencyList, equipmentIDPattern, equipmentNamePattern);
+                break;
+            case HVDC_LINE:
+                contingencies = getHvdcContingencyList(network, filtersContingencyList, equipmentIDPattern, equipmentNamePattern);
+                break;
+            case BUSBAR_SECTION:
+                contingencies = getBusbarSectionContingencyList(network, filtersContingencyList, equipmentIDPattern, equipmentNamePattern);
+                break;
+            case DANGLING_LINE:
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/actions/server/ContingencyListService.java:214]: We can't choose 'DANGLING_LINE' from the gui ...

**Review by FranckLecuyer — COMMENTED** [review]
Add the choice 'DANGLING_LINE' in the gui

File: src/main/java/org/gridsuite/actions/server/ContingencyListService.java (hunk 3 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/actions/server/ContingencyListService.java:214]: Ok, i have seen the addition in gridactions-app

**Review by FranckLecuyer — APPROVED** [review]

File: src/main/java/org/gridsuite/actions/server/dto/FilterContingencyListAttributes.java (hunk 2 above)
Comment (geofjamg) [reply src/main/java/org/gridsuite/actions/server/dto/FilterContingencyListAttributes.java:22]: You don't answer the question, This 2 classes are identical and used for the same pupose so why you don't keep just one (FilterContingencyList)?

File: src/main/java/org/gridsuite/actions/server/ContingencyListService.java [hunk 4]
```diff
@@ -109,13 +288,21 @@ void renameContingencyList(String name, String newName) {
         if (LOGGER.isDebugEnabled()) {
             LOGGER.debug("rename script contingency list '{}' to '{}'", sanitizeParam(name), sanitizeParam(newName));
         }
-        Optional<ScriptContingencyListEntity> optionalContingencyListEntity = repository.findByName(name);
-        if (optionalContingencyListEntity.isPresent()) {
-            repository.deleteByName(name);
+        Optional<ScriptContingencyListEntity> script = scriptContingencyListRepository.findByName(name);
+        Optional<FiltersContingencyListEntity> filters = filtersContingencyListRepository.findByName(name);
+        if (script.isPresent()) {
```
Comment (geofjamg) [inline src/main/java/org/gridsuite/actions/server/ContingencyListService.java:293]: Also maybe we can try to avoid the isPresent and use .map()

**Review by geofjamg — CHANGES_REQUESTED** [review]
Could you please look at the difference between regex and globbing. You have to choose between one the 2 solutions and not try to mix everything with something inconsistent.
- either you use regex so ".*" is the way to filter all
- either you choose another globbing system which support ".* (search for a standard lib for that)

**Review by geofjamg — CHANGES_REQUESTED** [review]
Could you please look at the difference between regex and globbing. You have to choose between one the 2 solutions and not try to mix everything with something inconsistent.
- either you use regex so ".*" is the way to filter all
- either you choose another globbing system which support ".* (search for a standard lib for that)

File: src/main/java/org/gridsuite/actions/server/dto/FilterContingencyListAttributes.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
```
Comment (geofjamg) [inline src/main/java/org/gridsuite/actions/server/dto/FilterContingencyListAttributes.java:22]: To remove

**Comment by geofjamg** [comment]
https://javapapers.com/java/glob-with-java-nio/
There is already a globbing function in the JDK used for path matching

File: src/main/java/org/gridsuite/actions/server/entities/FiltersContingencyListEntity.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+    private String equipmentId;
+
+    private String equipmentName;
+
+    private String equipmentType;
+
+    private double nominalVoltage;
+
+    private String nominalVoltageOperator;
```
Comment (geofjamg) [inline src/main/java/org/gridsuite/actions/server/entities/FiltersContingencyListEntity.java:35]: You can use an enum here with an annotation to describe mapping to text (see other micro services how it is done)

**Review by geofjamg — APPROVED** [review]

**Pull request merged**